examples/typescript-react-example/src/EnumsWithPayload.gen.tsx

```diff
@@ -13,6 +13,9 @@ export type payload = {readonly x: number, readonly y?: string};
 // tslint:disable-next-line:interface-over-type-literal
 export type withPayload = "a" | "bRenamed" | true | 20 | 0.5 | payload;
 
+// tslint:disable-next-line:interface-over-type-literal
+export type manyPayloads = ["one", number] | ["two", [string, string]];
```

Comment:
@cristianoc This is awesome!! 🎉 

One thing to consider is that Flow doesn't really support refining over tuples, unfortunately ([example](https://flow.org/try/#0PTAEAEDMBsHsHcBQiAuBPADgU1AQQE74CGaAakfgJZEB2KoAvKANor4CuWANKAEwDMAFgC6oAD4tIRaAGduoAOQAhIgBNQ+LAEdOMlAuEBuZOmygA8gCMAVlgDGKclVr0mAb1Az2du1hkyALlA2Th4AN2lOIIFBUABfcVAPLx8-QNApWXksQlh8IOU1DW1dfXjjRDtYGj1QAHMsR0icJgAKCOc6IKtbBydqOgBKIJp2AFtLHMYAPiTEUFBKSFB2igGUADoU339GBiYQrEG5hYXNFHZ8GlAO9eYARiN5+NAsLJPT88vrgAZjBbiiEBlWqtQaTWgnEYK1uLiCBGIZDWLmGoFGEymDFmbmeSxhyLozB+on2Bw4Rw+Z0a3xuBJQDyeANe7xxp2KFyuoD+z0BcSAA)). I wonder if there can be a different logic for TS and Flow, or if enums could be converted directly to objects instead of arrays?

Reply:
@jchavarri good point?
I think it’s better to avoid different representations for different backends, as one can later switch. Eg also with untyped.
Sounds like an object representation is the way to go.

Reply:
For what is worth, there's a PR with a fix in https://github.com/facebook/flow/pull/5048 so potentially Flow could support this tuple-based refinement too. But it's been opened for a year w/o feedback so who knows if / when it'll be fixed. 🤷‍♂️